x-pack/plugins/fleet/server/services/agents/unenroll.ts

```diff
@@ -56,11 +56,12 @@ export async function unenrollAgent(
 export async function unenrollAgents(
   soClient: SavedObjectsClientContract,
   esClient: ElasticsearchClient,
-  options: GetAgentsOptions
+  options: GetAgentsOptions & { force?: boolean }
 ) {
+  // start with all agents specified
   const agents = await getAgents(esClient, options);
 
-  // Filter to agents that are not already unenrolled, or unenrolling
+  // Filter to those not already unenrolled, or unenrolling
   const agentsEnrolled = agents.filter(
     (agent) => !agent.unenrollment_started_at && !agent.unenrolled_at
```

Comment:
There was one difference between the filtering logic of the two functions. `force*` only tested `unenrolled_at` vs this function which also checks `unenrollment_started_at`

```diff 
- (agent) => !agent.unenrolled_at;
+ (agent) => !agent.unenrollment_started_at && !agent.unenrolled_at
```

I believe this is the correct logic, but let me know if it should be different if `force: true`

Reply:
I think the difference was correct, as you could have to force unenroll an agent that started the enrollment before but never finished.